docs/using/permissions.md

```diff
@@ -0,0 +1,399 @@
+# Managing Permissions
+
+Wassette uses a fine-grained permission system to control what resources WebAssembly components can access. This page explains how to work with permissions in your day-to-day use of Wassette.
+
+## Overview
+
+Every component in Wassette runs in a secure sandbox with **deny-by-default** permissions. This means:
+
+- **No access by default**: Components cannot access files, networks, or environment variables unless explicitly granted
+- **Per-component policies**: Each component has its own independent permission set
+- **Runtime enforcement**: The WebAssembly sandbox blocks unauthorized access attempts
+
+## Permission Types
+
+Wassette supports four types of permissions:
+
+### Storage Permissions
+
+Control file system access for reading and writing files.
+
+**Example uses:**
+- Allow a component to read configuration files
+- Grant write access to output directories
+- Restrict access to specific workspace folders
+
+### Network Permissions
+
+Control outbound network access to specific hosts.
+
+**Example uses:**
+- Allow API calls to external services
+- Permit access to specific domains only
+- Restrict network egress for security
+
+### Environment Variable Permissions
+
+Control access to environment variables.
+
+**Example uses:**
+- Provide API keys to components
+- Share configuration via environment
+- Control access to sensitive credentials
+
+### Memory Permissions
+
+Set memory limits for components (future capability).
+
+**Example uses:**
+- Prevent resource exhaustion
+- Enforce quotas in multi-tenant environments
+
+## Granting Permissions
+
+The recommended way to grant permissions is through your AI agent when running Wassette as an MCP server. You can also use CLI commands for direct management, or define permissions in policy files.
+
+### Using MCP Built-in Tools (Recommended)
+
+When running Wassette as an MCP server, simply ask your AI agent to grant permissions in natural language:
+
+```text
+Please grant storage read and write permissions to the weather-tool for fs://workspace/
+```
+
+The agent will automatically use the appropriate built-in tool to apply the permission.
+
+**More examples:**
+
+```text
+Grant network access to api.weather.com for the weather-tool component
+```
+
+```text
+Allow the weather-tool to access the API_KEY environment variable
+```
+
+**Available MCP tools:**
+- `grant-storage-permission`: Grant file system access
+- `grant-network-permission`: Grant network access
+- `grant-environment-variable-permission`: Grant environment variable access
```

Comment:
Add a note that the even after the env var permissions are granted, the server still needs to be able to see those env vars. 

One way of doind that is to run `wassette secret set` to inject secrets into this. Another way is to run the server with the necessasry env vars